src/perform_step/rkc_perform_step.jl

```diff
@@ -460,3 +460,260 @@ end
   integrator.k[2] = integrator.fsallast
   integrator.u = u
 end
+
+function initialize!(integrator, cache::IRKCConstantCache)
+  @unpack uprev, p, t = integrator
+  @unpack f1, f2 = integrator.f
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(undef, integrator.kshortsize)
+  cache.du₁ = f1(uprev,p,t)
+  cache.du₂ = f2(uprev,p,t)
+  integrator.fsalfirst = cache.du₁ + cache.du₂
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::IRKCConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p,alg,fsalfirst = integrator
+  @unpack du₁,du₂,nlsolve = cache
+  @unpack f1, f2 = integrator.f
+  maxeig!(integrator, cache)
+  nlsolve!, nlcache = nlsolve, nlsolve.cache
+
+  # The the number of degree for Chebyshev polynomial
+  maxm = max(2,Int(floor(sqrt(integrator.opts.internalnorm(integrator.opts.reltol,t)/(10.0*eps(integrator.opts.internalnorm(uprev,t)))))))
+  mdeg = 1 + Int(floor(sqrt(1.54*dt*integrator.eigen_est + 1.0)))
+  if mdeg >= maxm
+    mdeg = maxm
+  end
+  mdeg = (mdeg < 50) ? 50 : mdeg
```

Comment:
@ChrisRackauckas I increased the number of stages manually by setting a minimum, and the answer is getting close to the exact answer. I think the `integrator.eigen_est` is being under calculated.

Reply:
good investigating. eigen_est is a very crude measure. Is there a safety factor on there to tweak?

Reply:
It would be great if we have an Arnoldi based algorithm.

Reply:
@ChrisRackauckas I don't think so! The problem is arising because the `eigen_est` is supposed to be calculated only for the non stiff part or `f.f2` in the `SplitODEFunction`, I tried to make those changes but It's not working.

Reply:
Ahh yes, do a `if typeof(integrator.f) <: SplitFunction` for that.